tests/service.rs

```diff
@@ -24,8 +24,38 @@ fn service_should_be_started_successfully() {
                 - key: kubernetes.io/arch
                   operator: Equal
                   value: stackable-linux
-        "},
+        "}),
     );
 
     client.verify_pod_condition(&pod, "Ready");
 }
+
+#[test]
+fn restart_after_ungraceful_shutdown_should_succeed() {
+    let client = TestKubeClient::new();
+
+    setup_repository(&client);
+
+    let pod_spec = with_unique_name(indoc! {"
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          name: agent-service-integration-test-restart
+        spec:
+          containers:
+            - name: nostop-service
+              image: nostop-service:1.0.1
+              command:
+                - nostop-service-1.0.1/start.sh
+          tolerations:
+            - key: kubernetes.io/arch
+              operator: Equal
+              value: stackable-linux
+          terminationGracePeriodSeconds: 2
```

Comment:
Wouldn't you want a large delay here for this test? Two seconds seems to be cutting it close to systemd terminating the service before the next pod is started which would create a false positive, no?

Reply:
`terminationGracePeriodSeconds` must be greater than the period between stopping the pod and creating the new systemd service. Additionally it must be lower than the timeout set in [`delete`](https://github.com/stackabletech/agent-integration-tests/blob/4e9a210dc3e4fc81d26143381b662ac854b3c016/tests/test/kube.rs#L273) which is currently 10 seconds. If these conditions are not met then the result is a false negative.

`terminationGracePeriodSeconds` is specific to pods whereas `delete` handles all types of resources. So the timeout in `delete` cannot be adapted implicitly.

`terminationGracePeriodSeconds` should be set as small as possible to minimize the runtime of the test case.

But I agree that two seconds could be too small for some systems. I will increase it to five seconds and make somehow the relation to the timeout in `delete` clear.

Reply:
`terminationGracePeriodSeconds` set to 5 seconds and default timeout extended accordingly